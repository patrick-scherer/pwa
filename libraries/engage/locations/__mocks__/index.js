export const FulfillmentSelector = jest.fn().mockReturnValue('FulfillmentSelector');
export const FulfillmentSheet = jest.fn().mockReturnValue('FulfillmentSheet');
