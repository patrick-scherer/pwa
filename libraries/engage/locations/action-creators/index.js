// @flow
export { default as errorLocations } from './errorLocations';
export { default as selectLocation } from './selectLocation';
export { default as selectGlobalLocation } from './selectGlobalLocation';
export { default as storeFormInput } from './storeFormInput';
export { default as storeFulfillmentMethod } from './storeFulfillmentMethod';
export { default as setUserSearchPostalCode } from './setUserSearchPostalCode';
export { default as setUserSearchCountryCode } from './setUserSearchCountryCode';
export { default as errorProductLocations } from './errorProductLocations';
export { default as receiveLocations } from './receiveLocations';
export { default as receiveProductLocations } from './receiveProductLocations';
export { default as requestLocations } from './requestLocations';
export { default as requestProductLocations } from './requestProductLocations';
export { default as submitReservationRequest } from './submitReservationRequest';
export { default as submitReservationSuccess } from './submitReservationSuccess';
export { default as submitReservationError } from './submitReservationError';
