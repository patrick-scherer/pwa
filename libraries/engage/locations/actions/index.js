export { default as fetchLocations } from './fetchLocations';
export { default as fetchProductLocations } from './fetchProductLocations';
export { default as submitReservation } from './submitReservation';
export { default as setUserSearchGeolocation } from './setUserSearchGeolocation';
