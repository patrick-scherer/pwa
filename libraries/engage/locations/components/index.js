// @flow
export { FulfillmentSelector } from './FulfillmentSelector';
export { FulfillmentSheet } from './FulfillmentSheet';
export { FulfillmentPathSelector } from './FulfillmentPathSelector';
export {
  StoreList,
  StoreOpeningHours,
  StorePhoneNumber,
  StoreAddress,
} from './StoreList';
export {
  CartItemProductChangeLocation,
  CartChangeFulfillmentMethod,
  CartContextMenuItemChangeLocation,
  CartContextMenuItemChangeFulfillment,
  CartContextMenuChangeFulfillment,
} from './Cart';

export { GlobalLocationSelector } from './GlobalLocationSelector';
export { GlobalLocationSwitcher } from './GlobalLocationSwitcher';
export { StoreFinder } from './StoreFinder';
