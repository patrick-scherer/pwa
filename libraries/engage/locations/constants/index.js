// @flow
// ACTION TYPES
export {
  REQUEST_LOCATIONS,
  RECEIVE_LOCATIONS,
  ERROR_LOCATIONS,
  SELECT_LOCATION,
  STORE_FORM_INPUT,
  STORE_FULFILLMENT_METHOD,
  REQUEST_PRODUCT_LOCATIONS,
  RECEIVE_PRODUCT_LOCATIONS,
  ERROR_PRODUCT_LOCATIONS,
  SUBMIT_RESERVATION_REQUEST,
  SUBMIT_RESERVATION_SUCCESS,
  SUBMIT_RESERVATION_ERROR,
} from './ActionTypes';

// PIPELINES
export {
  SHOPGATE_STOREFRONT_GET_PRODUCT_LOCATIONS,
  SHOPGATE_STOREFRONT_CREATE_ORDER,
  SHOPGATE_STOREFRONT_GET_LOCATIONS,
  SHOPGATE_STOREFRONT_GET_CART,
  SHOPGATE_STOREFRONT_UPDATE_CART,
} from './Pipelines';

// PORTALS
export {
  PRODUCT_FULFILLMENT_SELECTOR,
  PRODUCT_LOCATION_STOCK_INFO,
} from './Portals';

export const DIRECT_SHIP = 'DIRECT_SHIP';
export const IN_STORE_PICKUP = 'inStorePickup';
export const ROPIS = 'ROPIS';
export const QUICK_RESERVE = 'quickReserve';
export const MULTI_LINE_RESERVE = 'multiLineReserve';

export {
  STAGE_SELECT_STORE,
  STAGE_RESERVE_FORM,
  STAGE_RESPONSE_SUCCESS,
  STAGE_RESPONSE_ERROR,
  STAGE_FULFILLMENT_METHOD,
} from './Stages';
