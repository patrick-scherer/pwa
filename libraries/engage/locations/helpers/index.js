// @flow
export { default as createOrder } from './createOrder';
export { default as getAvailabilitySettings } from './getAvailabilitySettings';
export { default as checkRopeFulfillmentMethodsSupport } from './checkRopeFulfillmentMethodsSupport';
export { default as getDefaultRopeFulfillmentMethod } from './getDefaultRopeFulfillmentMethod';
export { isProductAvailable } from './productInventory';
