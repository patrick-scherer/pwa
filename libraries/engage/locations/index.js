// @flow

// ACTIONS
export { fetchProductLocations } from './actions';

// COMPONENTS
export {
  FulfillmentSelector,
  FulfillmentSheet,
  FulfillmentPathSelector,
  StoreList,
  StoreAddress,
  StoreOpeningHours,
  StorePhoneNumber,
  CartItemProductChangeLocation,
  CartChangeFulfillmentMethod,
  CartContextMenuItemChangeLocation,
  CartContextMenuItemChangeFulfillment,
} from './components';

// CONSTANTS
export {
  PRODUCT_FULFILLMENT_SELECTOR,
  PRODUCT_LOCATION_STOCK_INFO,
  DIRECT_SHIP,
  DIRECT_SHIP_LABEL,
  IN_STORE_PICKUP_BOPIS_LABEL,
  IN_STORE_PICKUP_ROPIS_LABEL,
  ROPIS,
  BOPIS,
  DEFAULT_ROPE_FULFILLMENT_METHOD,
  QUICK_RESERVE,
  MULTI_LINE_RESERVE,
  STAGE_SELECT_STORE,
  STAGE_RESERVE_FORM,
  STAGE_RESPONSE_SUCCESS,
  STAGE_RESPONSE_ERROR,
  STAGE_FULFILLMENT_METHOD,
} from './constants';

// SELECTORS
export {
  makeGetProductFulfillmentMethods,
  makeIsFulfillmentSelectorMethodEnabled,
  makeGetUserLocation,
  makeGetLocation,
  makeGetProductLocation,
  makeIsRopeProductOrderable,
  makeGetUserSearchCountryCode,
  makeGetUserSearchPostalCode,
  makeGetUserLocationFulfillmentMethod,
} from './selectors';

// STREAMS
export {
  submitReservationSuccess$,
} from './locations.streams';

// HOOKS
export {
  useFulfillmentState,
} from './locations.hooks';

// TYPES
export type {
  LocationAddress,
  Location,
  LocationsByIdState,
  LocationsByProductIdState,
  FulfillmentPath,
  UserLocationState,
  ReservationFormValues,
  UserLocationFulfillmentMethod,
  UserLocationLocationCode,
  UserFormInputState,
  LocationsState,
  SheetStage,
  SheetCallbackFn,
  SheetOpenParams,
  ReservationResponse,
  LocationOperationHours,
  LocationAware,
  OptionalLocationAware,
} from './locations.types';
