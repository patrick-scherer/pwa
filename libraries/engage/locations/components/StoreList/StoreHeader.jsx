// @flow
import React, { useCallback, useContext } from 'react';
import { Grid } from '@shopgate/engage/components';
import { FulfillmentContext } from '../../locations.context';
import { StoreContext } from './Store.context';
import { StoreDistance } from './StoreDistance';
import { StoreHoursToday } from './StoreHoursToday';
import { storeHeader, storeName } from './Store.style';

/**
 * Renders a single store headline.
 * @returns {JSX}
 */
export function StoreHeader() {
  const store = useContext(StoreContext);
  const { selectLocation } = useContext(FulfillmentContext);

  const handleClick = useCallback(() => {
    selectLocation(store);
  }, [selectLocation, store]);

  const { name, distance, unitSystem } = store;

  return (
    <div
      className={storeHeader}
      onClick={handleClick}
      onKeyDown={handleClick}
      role="button"
      tabIndex={0}
    >
      <Grid>
        <Grid.Item grow={1} className={storeName}>
          {name}
        </Grid.Item>
        <Grid.Item shrink={0}>
          <StoreDistance distance={distance} unitSystem={unitSystem} />
        </Grid.Item>
      </Grid>
      <StoreHoursToday hours={store.operationHours} />
    </div>
  );
}
