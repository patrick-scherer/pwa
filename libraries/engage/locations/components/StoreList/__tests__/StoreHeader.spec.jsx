import React from 'react';
import { mount } from 'enzyme';
import { FulfillmentContext } from '../../../locations.context';
import { isProductAvailable } from '../../../helpers';
import { StoreContext } from '../Store.context';
import { StoreHeader } from '../StoreHeader';

const context = {
  selectLocation: jest.fn(),
};

jest.mock('../../../helpers', () => ({
  isProductAvailable: jest.fn().mockReturnValue(true),
}));

describe('engage > locations > StoreList > StoreHeader', () => {
  const store = {
    code: 'store',
    name: 'SomeStore',
    distance: 0.02,
    unitSystem: 'metric',
    operationHours: {
      mon: '8:00am - 4:00pm',
      tue: '8:00am - 4:00pm',
      wed: '8:00am - 4:00pm',
      thu: '8:00am - 4:00pm',
      fri: '8:00am - 4:00pm',
      sat: '8:00am - 4:00pm',
      sun: '8:00am - 4:00pm',
    },
  };

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should render as expected', () => {
    const wrapper = mount((
      <FulfillmentContext.Provider value={context}>
        <StoreContext.Provider value={store}>
          <StoreHeader />
        </StoreContext.Provider>
      </FulfillmentContext.Provider>
    ));
    expect(wrapper).toMatchSnapshot();
  });

  it('should handle the store selection', () => {
    const wrapper = mount((
      <FulfillmentContext.Provider value={context}>
        <StoreContext.Provider value={store}>
          <StoreHeader />
        </StoreContext.Provider>
      </FulfillmentContext.Provider>
    ));
    wrapper.find('div[role="button"]').simulate('click');
    expect(context.selectLocation).toHaveBeenCalledWith({ ...store });
  });

  it('should not handle the store selection when the product is not available', () => {
    isProductAvailable.mockReturnValueOnce(false);
    const wrapper = mount((
      <FulfillmentContext.Provider value={context}>
        <StoreContext.Provider value={store}>
          <StoreHeader />
        </StoreContext.Provider>
      </FulfillmentContext.Provider>
    ));
    wrapper.find('div[role="button"]').simulate('click');
    expect(context.selectLocation).not.toHaveBeenCalled();
  });
});
