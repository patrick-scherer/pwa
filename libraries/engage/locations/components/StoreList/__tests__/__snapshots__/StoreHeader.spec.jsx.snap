// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`engage > locations > StoreList > StoreHeader should render as expected 1`] = `
<StoreHeader>
  <div
    className="css-1qplxu7"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="button"
    tabIndex={0}
  >
    <Grid
      className=""
      component="ul"
      wrap={false}
    >
      <ul
        className=" css-h7r55a"
      >
        <GridItem
          className="css-1qlb3ci"
          component="li"
          grow={1}
          shrink={1}
        >
          <li
            className="css-1qlb3ci css-1gs0ko2"
          >
            SomeStore
            <ResponsiveContainer
              appAlways={false}
              appOnly={false}
              breakpoint=">=sm"
              webAlways={false}
              webOnly={true}
            />
          </li>
        </GridItem>
        <ResponsiveContainer
          appAlways={true}
          appOnly={false}
          breakpoint="<sm"
          webAlways={false}
          webOnly={false}
        >
          <GridItem
            className=""
            component="li"
            grow={0}
            shrink={0}
          >
            <li
              className=" css-1l9a57c"
            >
              <StoreDistance
                distance={0.02}
                unitSystem="metric"
              >
                <span
                  className="css-jr9m0l"
                >
                  formats.distance.pattern
                </span>
              </StoreDistance>
            </li>
          </GridItem>
        </ResponsiveContainer>
        <ResponsiveContainer
          appAlways={false}
          appOnly={false}
          breakpoint=">=sm"
          webAlways={false}
          webOnly={true}
        />
      </ul>
    </Grid>
    <StoreHoursToday
      hours={
        Object {
          "fri": "8:00am - 4:00pm",
          "mon": "8:00am - 4:00pm",
          "sat": "8:00am - 4:00pm",
          "sun": "8:00am - 4:00pm",
          "thu": "8:00am - 4:00pm",
          "tue": "8:00am - 4:00pm",
          "wed": "8:00am - 4:00pm",
        }
      }
    >
      <div
        className="css-17tiibl"
      >
        locations.today_hours
      </div>
    </StoreHoursToday>
  </div>
</StoreHeader>
`;
